File: HwangSettong/week_2/144p.java
import java.util.Stack;
class Solution {
	static boolean solution(String s) {
		Stack<String> stack = new Stack<>();
		for (int i = 0; i < s.length(); i++) {
			if (s.charAt(i) == '(') {
				stack.push("(");
			} else {
				if (stack.isEmpty()) {
					return false;
				}
				stack.pop();
			}
		}
		if (!stack.isEmpty()) {
			return false;
		}
		return true;
	}
}

File: jane-kim-279/week_2/144p.py
def solution(s):
    stack = []
    for i in s:
        if i == "(":
            stack.append(i)
        else:
            if len(stack) != 0 and stack[-1] == "(": stack.pop()
            else: stack.append(i)
    return len(stack) == 0
